controllers/auth/index.js

```diff
@@ -1,21 +1,136 @@
 import express from "express"
+import { createUser } from "../../services/user/index.js"
+import { createToken } from "../../services/jwt/index.js";
+import * as z from "zod";
 
 const router = express.Router();
 
 // TODO: develop the APIs and add the swagger documentation here using JSDoc syntax
 
-router.post("/login", (req, res) => {
+const userSchema = z.object({
```

Comment:
`userSchema` isn't a valid name in this case, since you're using this for the register controller, rename it to `registerBodyPayload`. Since `userSchema` would create an impression of DB model